src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -261,17 +265,9 @@ private bool IsOpenProperty(ODataProperty property)
 
         private void WriteUntypedValue(ODataUntypedValue untypedValue)
         {
-            if (!this.MessageWriterSettings.ThrowOnUndeclaredPropertyForNonOpenType)
-            {
-                this.JsonWriter.WriteName(this.currentPropertyInfo.WireName);
-                this.jsonLightValueSerializer.WriteUntypedValue(untypedValue);
-                return;
-            }
-
-            Debug.Assert(
-                this.MessageWriterSettings.ThrowOnUndeclaredPropertyForNonOpenType,
-                "this.MessageWriterSettings.ThrowOnUndeclaredPropertyForNonOpenType");
-            throw new ODataException(ODataErrorStrings.ValidationUtils_PropertyDoesNotExistOnType(this.currentPropertyInfo.PropertyName, this.currentPropertyInfo.MetadataType.OwningType.FullTypeName()));

```

Comment:
Remove this is a breaking change?

Reply:
We do the check before this, in WriteProperty (which is where we should check the property name, not in WriteUntypedValue). There we correctly check to see if the type is no-open.  So this code is only reached after we've verified that that it's valid.

So the only behavioral change is, if the property is undeclared and the type is open we no longer throw the exception for untyped values.  This enables a code path that was previously an error (but shouldn't have been).   

---
In reply to: [154829236](https://github.com/OData/odata.net/pull/1010#discussion_r154829236) [](ancestors = 154829236)